src/hotspot/share/gc/shenandoah/heuristics/shenandoahOldHeuristics.cpp

```diff
@@ -310,37 +309,29 @@ void ShenandoahOldHeuristics::prepare_for_old_collections() {
 
   const size_t num_regions = heap->num_regions();
   size_t cand_idx = 0;
-  size_t total_garbage = 0;
   size_t immediate_garbage = 0;
   size_t immediate_regions = 0;
   size_t live_data = 0;
 
   RegionData* candidates = _region_data;
   for (size_t i = 0; i < num_regions; i++) {
     ShenandoahHeapRegion* region = heap->get_region(i);
-    if (!_old_generation->contains(region)) {
+    if (!region->is_old()) {
       continue;
     }
 
     size_t garbage = region->garbage();
     size_t live_bytes = region->get_live_data_bytes();
-    total_garbage += garbage;
     live_data += live_bytes;
 
-    // Only place regular regions into the candidate set
-    if (region->is_regular()) {
-      if (!region->has_live()) {
-        assert(!region->is_pinned(), "Pinned region should have live (pinned) objects.");
-        region->make_trash_immediate();
-        immediate_regions++;
-        immediate_garbage += garbage;
-      } else {
-        region->begin_preemptible_coalesce_and_fill();
-        candidates[cand_idx]._region = region;
-        candidates[cand_idx]._u._live_data = live_bytes;
-        cand_idx++;
-      }
+    if (region->is_trash()) {
+      // Count humongous objects made into trash here.
+      immediate_regions++;
+      immediate_garbage += garbage;
     } else if (region->is_humongous_start()) {
+      // This will handle humongous start regions whether they are also pinned, or not.
```

Comment:
Since we've already handled region->is_trash() above, do we need this code?  If we want to preserve the assert(!is_pinned), we can insert that into the is_trash() code above.  Right?

Reply:
I didn't change the implementation of `is_humongous_start`, so it still includes pinned humongous objects.

Reply:
My question was misguided.  Ignore it.  (I was confused when I was first reading this PR.)